public/app/features/serviceaccounts/ServiceAccountPage.tsx

```diff
@@ -55,6 +93,45 @@ const ServiceAccountPageUnconnected = ({
     deleteServiceAccountToken(parseInt(match.params.id, 10), key.id!);
   };
 
+  const onCreateToken = () => {
+    createServiceAccountToken(
+      serviceAccount.id,
+      {
+        name: newTokenName,
+        role: OrgRole.Viewer,
+      },
+      onTokenCreated
+    );
+  };
+
+  const onTokenCreated = (key: string) => {
+    setNewToken(key);
+  };
```

Comment:
Yes, feel the same. It's might be anonymous function since it's short:
```ts
const onCreateToken = () => {
    createServiceAccountToken(
      serviceAccount.id,
      {
        name: newTokenName,
        role: OrgRole.Viewer,
      },
      (key) => setNewToken(key)
    );
  };
```

Reply:
You don't really need this function tbh, since it does exactly the same as `setNewToken` it can be used directly instead.